aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2426,9 +2433,19 @@ impl AptosVM {
     ) -> Result<Option<u64>, VMStatus> {
         match payload {
             TransactionPayload::EntryFunction(entry_func) => {
-                if let Some(gas_amount) = self.randomness_api_v0_required_deposit.gas_amount {
-                    if has_randomness_attribute(resolver, session, entry_func).unwrap_or(false) {
-                        if gas_amount != u64::from(txn_metadata.max_gas_amount) {
+                if let Some(annotation) =
+                    get_randomness_annotation(resolver, session, entry_func).unwrap_or(None)
```

Comment:
i don't know if changing it now breaks compatibility... current code has been shipped with 1.12.

Reply:
anyway, why is it not fine to have this function focus on required deposit calculation?
if module loading failed, other existing steps should catch it anyway?

Reply:
i thought about it, but putting related configs close to each other also has its benefit...

Reply:
I don't like relying on "other existing steps"... we do not have clear semantics of what are the next steps, and this can bite us in the future. It is ok here, I guess...